### DISCUSSION TITLE
Current roadmap/development status of TypeGraphQL

### DISCUSSION THREAD

**Discussion by itpropro** [MichalLytek/type-graphql#1271] [General]
As there was no release since October 2021 and we are still on RC and the GitHub project “TypeGraphQL vNext” doesn’t seem very up to date, it would be great to get some insights into the active development.
In contrary to most Open Source projects, there is also no dev branch to get insights or a chance to contribute to the development.
I think a proper roadmap and some more regular releases would help the project a lot, as I often get questions why I am using a “dead” product and have to explain to people that it is alive and can still be used.

**Comment by itpropro** [comment]
Just to add to that, the latest "stable" release is from November 2020, the last code change from over half a year ago and there are Pull Requests that are open for over a year now. Making the current status of the development more transparent and enable more collaboration opportunities on the project, like having community maintainers and open development on some kind of dev branch, would help getting back on track with all the brilliant stuff in the backlog :)

**Comment by MichalLytek** [comment]
> there is also no dev branch to get insights or a chance to contribute to the development

`master` serves that role, I work in a simple Github Flow with PRs merged directly to master.
Then I release the RC and after stabilizing the stable version on npm.

> I often get questions why I am using a “dead” product and have to explain to people that it is alive and can still be used.

The project is not dead, it's mature. The RC release contains all the latest fixes and there's no major pending bug report.
So there's no need for everyday commit. But a lot of improvement can be made in the near future.

Generally speaking, I currently don't have enough time to work on some big features.
Last months I was really busy with renovating my new house, I expect to settle down in a few months.
But in a few weeks I should be able to finish v16 upgrade and release the v2 RC.

**Reply by itpropro** [reply]
Thanks for your reply @MichalLytek! 
I get that the project uses a simple Gitflow and that the project is mature. It is just not recognizable on first or second sight when you discover the project. 
I would call most of the Open Source projects I use mature, but they still have at least a simple roadmap and recent releases (even if it's bugfix releases), because from my experience, just because it is mature, doesn't mean there is nothing to do (often the opposite is the case, this project also had ~80 open issues and 12 open PRs) and frequent releases is one of the things people use to identify mature and active projects.
Maybe these are not things that would make the projects codebase better, but it would make the adoption easier and would raise the visibility for TypeGraphQL.
I am using TypeGraphQL regularly and it saved me a lot of time, reduced maintenance overhead and often cleaned up the codebase. So I will continue using and adopting it anyway. This is just some feedback I received from customers (medium to big sized companies as well as freelancers) that I wanted to share.

**Reply by MichalLytek** [reply]
I appreciate the feedback and agree with that situation 😉 

I am a sole developer, have my daily job, really busy after work, so hard to find some time for OSS.
Apart from the companies I work for, I receive about ~$435 of monthly support from the others by OpenCollective.
Hard to reduce my daily job and live from sponsorships 😬 
Some contribution from the "medium to big sized companies" would help a lot.

**Reply by itpropro** [reply]
I’m completely with you on that contribution part, if anything, I try to convince people to contribute to all the projects they are benefiting from.
That’s why I brought that up, as many people I know just look at the release cadence, frequency and progress to decide if they sponsor. So if there is any way to get some of the hotfixes out as new versions, even with minor changes, that wouldn’t produce too much work for you, it would also help the project getting more funds.
For me it’s about getting TypeGraphQL the attention it deserves, which should also result in better funding for the project and the time you are investing instead of working for your employer/customer.
I think there is also a certain lack of external developers working on the project compared with other Open Source projects, maybe there is something that could be improved with a more detailed contribution guide or similar to easier enable other people to add features/fix bugs, which would also get some work off your shoulders :)
